### PR TITLE
Add playrec.py example

### PR DESCRIPTION
**Pull request by fhchl**
As discussed in #162.

Gotta quickly test this Monday with a device to see if there didn't sneak any mistake in.

### CODE REVIEW

**Review by zhindes — APPROVED** [review]
Looks great. Let me know when you've done your testing, and I can merge.

File: nidaqmx_examples/playrec.py [hunk 1]
```diff
@@ -0,0 +1,83 @@
+"""Simultaneous read and write with NI USB-4431 or similar device."""
+
+import numpy as np
+import nidaqmx as ni
+from nidaqmx.constants import WAIT_INFINITELY
+
+
+def query_devices():
```
Comment (zhindes) [inline nidaqmx_examples/playrec.py:8]: This function isn't used.

File: nidaqmx_examples/playrec.py [hunk 2]
```diff
@@ -0,0 +1,83 @@
...
+    driver.major_version, driver.minor_version, driver.update_version))
+
+  for device in local.devices:
+    print('Device Name: {0}, Product Category: {1}, Product Type: {2}'.format(
+      device.name, device.product_category, device.product_type))
+
+
+def playrec(
+  data, samplerate=96000, input_mapping=['Dev2/ai0', 'Dev2/ai1', 'Dev2/ai2'],
+  output_mapping=['Dev2/ao0']
```
Comment (zhindes) [inline nidaqmx_examples/playrec.py:22]: default parameters for these are too application-specific. can you remove them and pass them in __main__?

File: nidaqmx_examples/playrec.py [hunk 3]
```diff
@@ -0,0 +1,83 @@
...
+    write_task.start()  # write_task doesn't start at read_task's start_trigger
+                        # without this
+    indata = read_task.read(nsamples, timeout=WAIT_INFINITELY) # do not time out
+                                                               # for long inputs
+
+  return np.asarray(indata).T
+
+
+if __name__ == '__main__':
+  # excite through one output and record at three inputs
```
Comment (zhindes) [inline nidaqmx_examples/playrec.py:81]: this comment will make more sense when you remove the default params and pass them in

File: nidaqmx_examples/playrec.py [hunk 4]
```diff
@@ -0,0 +1,83 @@
...
+    Output device channels
+
+  Returns
+  -------
+  ndarray, shape (nsamples, len(input_mapping))
+    Recorded data
+
+  """
+  max_out_range = 3.5 # output range of USB-4431
+  max_in_range = 10   # input range of USB-4431
```
Comment (zhindes) [inline nidaqmx_examples/playrec.py:44]: you might be able to query these from `system.device.ai_voltage_rngs` and `system.device.ao_voltage_rngs`

File: nidaqmx_examples/playrec.py [hunk 5]
```diff
@@ -0,0 +1,83 @@
...
+      f"(+-{max_out_range:.2f}).")
+
+  outdata = np.asarray(outdata).T
+  nsamples = outdata.shape[1]
+
+  with ni.Task() as read_task, ni.Task() as write_task:
+    for o in output_mapping:
+      aochan = write_task.ao_channels.add_ao_voltage_chan(o)
+      aochan.ao_max = max_out_range
+      aochan.ao_min = -max_out_range
```
Comment (zhindes) [inline nidaqmx_examples/playrec.py:58]: these are input params on `add_ao_voltage_chan`: `min_val`, `max_val`

File: nidaqmx_examples/playrec.py [hunk 6]
```diff
@@ -0,0 +1,83 @@
...
+    for i in input_mapping:
+      aichan = read_task.ai_channels.add_ai_voltage_chan(i)
+      aichan.ai_min = -max_in_range
+      aichan.ai_max = max_in_range
```
Comment (zhindes) [inline nidaqmx_examples/playrec.py:62]: these are input params on `add_ai_voltage_chan`: `min_val`, `max_val`

File: nidaqmx_examples/playrec.py [hunk 7]
```diff
@@ -0,0 +1,83 @@
...
+                                      samps_per_chan=nsamples)
+
+    # trigger write_task as soon as read_task starts
+    write_task.triggers.start_trigger.cfg_dig_edge_start_trig(
+            read_task.triggers.start_trigger.term)
+    write_task.write(outdata, auto_start=False)
+    write_task.start()  # write_task doesn't start at read_task's start_trigger
+                        # without this
+    indata = read_task.read(nsamples, timeout=WAIT_INFINITELY) # do not time out
+                                                               # for long inputs
```
Comment (zhindes) [inline nidaqmx_examples/playrec.py:75]: you switched to trailing comments here. do you mind moving them to their own line like the rest of the file for consistency?

**Review by zhindes — CHANGES_REQUESTED** [review]

File: nidaqmx_examples/playrec.py (hunk 1 above)
Comment (fhchl) [reply nidaqmx_examples/playrec.py:8]: Changed example slightly such that this function is actually useful here.

File: nidaqmx_examples/playrec.py (hunk 2 above)
Comment (fhchl) [reply nidaqmx_examples/playrec.py:22]: Good idea.

File: nidaqmx_examples/playrec.py (hunk 3 above)
Comment (fhchl) [reply nidaqmx_examples/playrec.py:81]: Agreed.

File: nidaqmx_examples/playrec.py (hunk 4 above)
Comment (fhchl) [reply nidaqmx_examples/playrec.py:44]: Good idea.

File: nidaqmx_examples/playrec.py (hunk 5 above)
Comment (fhchl) [reply nidaqmx_examples/playrec.py:58]: I was wondering why these are not automatically set when passing a device to `add_ai_voltage_chan`. Seems like they must be passed explicitly.

File: nidaqmx_examples/playrec.py (hunk 6 above)
Comment (fhchl) [reply nidaqmx_examples/playrec.py:62]: see above

File: nidaqmx_examples/playrec.py (hunk 7 above)
Comment (fhchl) [reply nidaqmx_examples/playrec.py:75]: Yes.

**Comment by fhchl** [comment]
Tested the current example which works fine and throws the value error when the output data is to large.

**Review by zhindes — APPROVED** [review]

**Pull request merged**